test/core/super_component/test_utils.py

```diff
@@ -71,25 +129,45 @@ def test_complex_nested_types(self):
         target_type = List[List[int]]
 
         # With unwrap_nested=True
-        assert _is_compatible(complex_type, target_type)
-        assert _is_compatible(target_type, complex_type)
+        is_compat, common = _is_compatible(complex_type, target_type)
+        assert is_compat
+        # Handle both typing.List and built-in list notation
+        assert common == List[List[int]] or common == list[list[int]]
+
+        is_compat, common = _is_compatible(target_type, complex_type)
+        assert is_compat
+        assert common == List[List[int]] or common == list[list[int]]
```

Comment:
@mdrazak2001 why do we need the `or` here? Currently it would be preferable if common was `List[List[int]]`